src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/hmi/unsubscribe_button_request.cc

```diff
@@ -0,0 +1,159 @@
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "sdl_rpc_plugin/commands/hmi/unsubscribe_button_request.h"
+#include "application_manager/event_engine/event_dispatcher_impl.h"
+#include "application_manager/message_helper.h"
+#include "application_manager/resumption/resume_ctrl.h"
+
+namespace sdl_rpc_plugin {
+using namespace application_manager;
+
+namespace commands {
+SDL_CREATE_LOG_VARIABLE("Commands")
+
+namespace hmi {
+
+UnsubscribeButtonRequest::UnsubscribeButtonRequest(
+    const application_manager::commands::MessageSharedPtr& message,
+    ApplicationManager& application_manager,
+    rpc_service::RPCService& rpc_service,
+    HMICapabilities& hmi_capabilities,
+    policy::PolicyHandlerInterface& policy_handle)
+    : RequestToHMI(message,
+                   application_manager,
+                   rpc_service,
+                   hmi_capabilities,
+                   policy_handle)
+    , EventObserver(application_manager.event_dispatcher()) {
+  hmi_apis::Common_ButtonName::eType button_name =
+      static_cast<hmi_apis::Common_ButtonName::eType>(
+          (*message_)[app_mngr::strings::msg_params]
+                     [app_mngr::strings::button_name]
+                         .asInt());
+
+  button_name_ = button_name;
+}
+
+UnsubscribeButtonRequest::~UnsubscribeButtonRequest() {
+  unsubscribe_from_event(hmi_apis::FunctionID::Buttons_UnsubscribeButton);
+}
+
+void UnsubscribeButtonRequest::Run() {
+  SDL_LOG_AUTO_TRACE();
+
+  subscribe_on_event(hmi_apis::FunctionID::Buttons_UnsubscribeButton,
+                     correlation_id());
+
+  ApplicationSharedPtr app =
+      application_manager_.application_by_hmi_app(application_id());
+
+  if (!app) {
+    SDL_LOG_ERROR("Application for connection key: " << application_id()
+                                                     << " was not found");
+    return;
+  }
+
+  app->AddPendingButtonUnsubscription(correlation_id(), button_name_);
+
+  SendRequest();
+}
+
+void UnsubscribeButtonRequest::onTimeOut() {
+  SDL_LOG_AUTO_TRACE();
+  application_manager_.updateRequestTimeout(
```

Comment:
Please double check If we really need this row

Reply:
removed in 2529ff7558e0a8cc1429308d019fd9e9fd7b57ca